vaadin-date-time-picker-flow-parent/vaadin-date-time-picker-flow/src/main/java/com/vaadin/flow/component/datetimepicker/DateTimePicker.java

```diff
@@ -327,6 +402,31 @@ public DateTimePicker(LocalDateTime initialDateTime, Locale locale) {
         setLocale(locale);
     }
 
+    private void addValidationListeners() {
+        getElement().addEventListener("change", e -> {
+            // No need to validate since it will be validated once the
+            // programmatically set value is updated on the client
+            if (pendingInputElementValueSyncs == 0) {
+                validate(true);
+            }
+        });
+        getElement().addEventListener("unparsable-change", e -> {
+            // No need to validate since it will be validated once the
+            // programmatically set value is updated on the client
+            if (pendingInputElementValueSyncs == 0) {
+                validate(true);
+            }
+        });
+
+        getElement().addEventListener("value-programmatically-set", e -> {
+            // Validate only for the final input element value sync caused by
+            // programmatically setting values
+            if (--pendingInputElementValueSyncs == 0) {
+                validate(true);
+            }
+        });
```

Comment:
Could you please explain why this additional event is necessary? What scenario is it intended to cover?